api/net/Areas/Subscriber/Controllers/ReportController.cs

```diff
@@ -320,23 +331,31 @@ public async Task<IActionResult> Generate(int id, [FromQuery] bool regenerate =
                 return c;
             });
             currentInstance.ContentManyToMany.AddRange(newContent);
-            _reportInstanceService.UpdateAndSave(currentInstance);
+            currentInstance = _reportInstanceService.UpdateAndSave(currentInstance);
             instances = _reportService.GetLatestInstances(id, user.Id);
+            currentInstance.ContentManyToMany.Clear();
+            currentInstance.ContentManyToMany.AddRange(_reportInstanceService.GetContentForInstance(currentInstance.Id));
+            report.Instances.Clear();
+            report.Instances.Add(currentInstance);
+            report.Instances.AddRange(instances.Where(i => i.Id != currentInstance.Id));
         }
         else if (currentInstance == null || currentInstance.SentOn.HasValue)
         {
             // Generate a new instance of this report if there is no current instance, or if it has already been sent.
             currentInstance = await _reportHelper.GenerateReportInstanceAsync(new Services.Models.Report.ReportModel(report, _serializerOptions), user.Id);
-            _reportInstanceService.AddAndSave(currentInstance);
+            currentInstance = _reportInstanceService.AddAndSave(currentInstance);
             instances = _reportService.GetLatestInstances(id, user.Id);
+            currentInstance.ContentManyToMany.Clear();
+            currentInstance.ContentManyToMany.AddRange(_reportInstanceService.GetContentForInstance(currentInstance.Id));
```

Comment:
To improve performance content is not automatically returned by a report instance, it needs to be fetched separately.